tests/core/mempool/test_mempool_manager.py

```diff
@@ -534,3 +535,89 @@ async def test_get_items_not_in_filter() -> None:
+        g1 = G1Element()
+        for _ in range(2436):
+            conditions.append([ConditionOpcode.AGG_SIG_UNSAFE, g1, IDENTITY_PUZZLE_HASH])
+        expected_cost1 = uint64(5498561056)
+    else:
+        expected_cost1 = uint64(2945056)
+    # We're leaving a small fee in this one
+    conditions.append([ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, coin1_amount - 42])
+    sb1, _, result = await generate_and_add_spendbundle(mempool_manager, conditions, coin1)
+    assert result == (expected_cost1, MempoolInclusionStatus.SUCCESS, None)
+    spend_bundles, cost_sum, additions, removals = mempool_manager.process_mempool_items(
+        item_inclusion_filter=inclusion_filter
+    )
+    if inclusion_filter == never:
+        # Make sure this mempool item got skipped by item_inclusion_filter
+        assert len(spend_bundles) == 0
+        assert cost_sum == 0
+        assert len(additions) == 0
+        assert len(removals) == 0
+    elif inclusion_filter == always:
+        # Make sure this mempool item was included by item_inclusion_filter
+        assert spend_bundles == [sb1]
+        assert cost_sum == expected_cost1
+        assert additions == [Coin(coin1.name(), IDENTITY_PUZZLE_HASH, coin1_amount - 42)]
+        assert removals == [coin1]
+        # Create a second item that when combined with the first one we exceed the maximum fee/block clvm cost
+        if test_for == "max_fee":
+            conditions = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 2]]
+            expected_cost2 = uint64(2957056)
+        else:
+            conditions = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, coin2_amount - 2]]
+            expected_cost2 = uint64(3065056)
+        sb2, _, result = await generate_and_add_spendbundle(mempool_manager, conditions, coin2)
+        assert result == (expected_cost2, MempoolInclusionStatus.SUCCESS, None)
+        spend_bundles, cost_sum, additions, removals = mempool_manager.process_mempool_items(
+            item_inclusion_filter=inclusion_filter
+        )
+        if test_for == "normal_path":
+            assert spend_bundles == [sb1, sb2]
+            assert cost_sum == expected_cost1 + expected_cost2
+            assert additions == [
+                Coin(coin1.name(), IDENTITY_PUZZLE_HASH, coin1_amount - 42),
+                Coin(coin2.name(), IDENTITY_PUZZLE_HASH, coin2_amount - 2),
+            ]
+            assert removals == [coin1, coin2]
+        else:
+            # The second item has a higher FPC so it should get picked first
+            assert spend_bundles == [sb2]
+            # The first item hits the maximum fee/block clvm cost and gets skipped
+            assert cost_sum == expected_cost2
+            if test_for == "max_cost":
+                assert additions == [Coin(coin2.name(), IDENTITY_PUZZLE_HASH, coin2_amount - 2)]
+            elif test_for == "max_fee":
+                assert additions == [Coin(coin2.name(), IDENTITY_PUZZLE_HASH, 2)]
```

Comment:
```suggestion
                assert additions == [Coin(coin2.name(), IDENTITY_PUZZLE_HASH, 2)]
            else:
                assert False, "not all cases covered"
```

Reply:
In case someone adds more later.

Reply:
Ah thanks, that was why I was explicit with the elif but this is better.